### PR TITLE
Draft app

### PR DESCRIPTION
**Pull request by Romeo1928**

### CODE REVIEW

File: src/app/formsSlice.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  },
+);
+
+// Создаем reducer с помощью formsSlice
+export const formsReducer = formsSlice.reducer;
+
+// Создаем actions на основе reducers из среза формы (лучше деструктуризацией)
+export const { setDataForms } = formsSlice.actions;
+
+export const formsActions = formsSlice.actions;
```
Comment (suveiko) [inline src/app/formsSlice.ts:71]: remove unused export

File: src/app/hooks.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { useDispatch, useSelector, TypedUseSelectorHook } from 'react-redux';
+import type { RootState, AppDispatch } from './store';
+
+// нужны для корректной работы с hooks useSelector, useDispatch
```
Comment (suveiko) [inline src/app/hooks.ts:4]: remove comments

File: src/components/Form.tsx [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+        .required('обязательное поле'),
+    ),
+    radio: yup.string().required('обязательное поле'),
+    checkbox: yup.array().min(1, 'обязательное поле'),
+
+    about: yup
+      .string()
+      .min(2, 'Поле обязательно к заполнению')
+      .max(200, 'Максимум 200 символов'),
+  });
```
Comment (suveiko) [inline src/components/Form.tsx:79]: move form outside of the component

File: src/components/Form.tsx [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+
+      about: '',
+    },
+
+    validate: yupResolver(validateSchema),
+  });
+
+  const handleSubmit = (values: typeof form.values) => {
+    // debugger;
+    console.log(values);
```
Comment (suveiko) [inline src/components/Form.tsx:104]: remove log

File: src/components/Form.tsx [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+  const prevStep = () => {
+    if (active === 0) {
+      return navigate('/');
+    }
+    setActive((prev) => prev - 1);
+  };
+
+  // Advantages
+  const fields = form.values.advantages.map((_, index) => (
+    <Group key={index} mt="xs">
```
Comment (suveiko) [inline src/components/Form.tsx:128]: dont use index as key

https://legacy.reactjs.org/docs/lists-and-keys.html

File: src/components/Form.tsx [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+  const handleSubmit = (values: typeof form.values) => {
+    // debugger;
+    console.log(values);
+    dispatch(setDataForms(values));
+    dispatch(formATh());
+  };
+
+  // STEPs
+  const nextStep = () => {
+    // console.log(form.values);
```
Comment (suveiko) [inline src/components/Form.tsx:111]: remove

File: src/components/Form.tsx [hunk 7]
```diff
@@ -0,0 +1,245 @@
...
+        checked={el.status}
+        {...form.getInputProps(`checkbox.${index}.status`)}
+      />
+    </Group>
+  ));
+
+  // console.log(form.values);
+
+  return (
+    <Box maw={680} p={'60px 0 80px 110px'}>
```
Comment (suveiko) [inline src/components/Form.tsx:155]:
```suggestion
    <Box maw={680} p='60px 0 80px 110px'>
```

File: src/components/MainPage.tsx [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+      href: '',
+      text: 'Resume',
+    },
+  ];
+
+  // yup
+  const validateSchema = yup.object().shape({
+    phone: yup.string().required('Обязательное поле'),
+    email: yup.string().required('Обязательное поле').email('Invalid email'),
+  });
```
Comment (suveiko) [inline src/components/MainPage.tsx:65]: move outside of the components

File: src/components/MainPage.tsx [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+      <div style={headerContainer}>
+        <div style={{ display: 'flex' }}>
+          <div style={avatar}>
+            <div>РР</div>
+          </div>
+          <div>
+            <h2>Роман Романов</h2>
+            <ul style={linksStyle}>
+              {links.map((item, index) => (
+                <li style={{ marginRight: '20px' }} key={index}>
```
Comment (suveiko) [inline src/components/MainPage.tsx:101]: dont use index as key

File: src/components/MainPage.tsx [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+                  <img src={item.icon} alt="" />
```
Comment (suveiko) [inline src/components/MainPage.tsx:102]:
```suggestion
                  <img src={item.icon} alt="icon" />
```

File: src/components/MainPage.tsx [hunk 11]
```diff
@@ -0,0 +1,132 @@
...
+                  <a href={item.href} target="blank">
```
Comment (suveiko) [inline src/components/MainPage.tsx:103]:
```suggestion
                  <a href={item.href} target="_blank">
```

File: src/components/MainPage.tsx [hunk 12]
```diff
@@ -0,0 +1,132 @@
...
+  const handleSubmit = (values: typeof form.values) => {
+    navigate('/step');
+    dispatch(setDataForms(values));
+    // console.log(values);
+  };
+
+  return (
+    <div style={container}>
+      <div style={headerContainer}>
+        <div style={{ display: 'flex' }}>
```
Comment (AdammmVV) [inline src/components/MainPage.tsx:93]: do not use inline style, use mantine components

File: src/components/MainPage.tsx [hunk 13]
```diff
@@ -0,0 +1,132 @@
...
+    },
+
+    validate: yupResolver(validateSchema),
+    //   validate: {
+    //     phone: (value) =>
+    //       /^\+7 \(\d{3}\) \d{3}-\d{2}-\d{2}$/.test(value)
+    //         ? null
+    //         : 'Неправильный формат номера телефона',
+    //     email: (value) => (/^\S+@\S+$/.test(value) ? null : 'Invalid email'),
+    //   },
```
Comment (AdammmVV) [inline src/components/MainPage.tsx:80]: remove unnecessary comments here and all the app

File: src/components/MainPage.tsx [hunk 14]
```diff
@@ -0,0 +1,132 @@
...
+const inputNumber: CSSProperties = {
+  marginBottom: '25px',
+};
+const linksStyle: CSSProperties = {
+  marginLeft: '-40px',
+  display: 'flex',
+  flexWrap: 'wrap',
+  gap: '16px',
+  listStyle: 'none',
+};
```
Comment (AdammmVV) [inline src/components/MainPage.tsx:38]: use mpdule.css or put it in separate file

File: src/components/MainPage.tsx [hunk 15]
```diff
@@ -0,0 +1,132 @@
...
+      icon: '',
+      href: '',
+      text: 'GitHub',
+    },
+    {
+      icon: '',
+      href: '',
+      text: 'Resume',
+    },
+  ];
```
Comment (AdammmVV) [inline src/components/MainPage.tsx:59]: put it in config

File: src/components/Form.tsx [hunk 16]
```diff
@@ -0,0 +1,245 @@
...
+    // console.log(form.values);
+    setActive((current) => {
+      if (form.validate().hasErrors) {
+        return current;
+      }
+      return current < 3 ? current + 1 : current;
+    });
+  };
+  const prevStep = () => {
+    if (active === 0) {
```
Comment (AdammmVV) [inline src/components/Form.tsx:120]: do not use magic numbers, use Enums for it `0, 1...`

File: src/components/Form.tsx [hunk 17]
```diff
@@ -0,0 +1,245 @@
...
+              style={{ width: '300px', height: '44px' }}
+              label="Sex"
+              placeholder="Не выбрано"
+              data={['man', 'woman']}
+              {...form.getInputProps('sex')}
+            />
+          </Stepper.Step>
+          {/*// 2 - Advantages*/}
+          <Stepper.Step>
+            <Group mt="65px" style={{ display: 'block' }}>
```
[the remark this answers is not in the document]
Comment (Romeo1928) [reply src/components/Form.tsx:193]: fields, checkbox, radio - as columns

File: src/api/form.api.ts [hunk 18]
```diff
@@ -0,0 +1,27 @@
+import axios from 'axios';
+
+const instance = axios.create({
+  baseURL: 'https://api.sbercloud.ru/',
+});
+
+export const formAPI = {
+  form(data: IForm) {
```
[the remark this answers is not in the document]
Comment (Romeo1928) [reply src/api/form.api.ts:8]: formAPI?

File: src/components/MainFormPage/MainFormPage.tsx [hunk 19]
```diff
@@ -0,0 +1,83 @@
...
+  });
+
+  const handleSubmit = (values: typeof form.values) => {
+    navigate(ROUTES.STEP);
+    dispatch(setDataForms(values));
+  };
+
+  return (
+    <Container style={container}>
+      <div style={headerContainer}>
```
Comment (AdammmVV) [inline src/components/MainFormPage/MainFormPage.tsx:39]: try use component [Box](https://mantine.dev/core/box/#usage) and [mantine styles ](https://mantine.dev/styles/style-props/)

File: src/components/MainFormPage/MainFormPage.tsx [hunk 20]
```diff
@@ -0,0 +1,83 @@
...
+            <ul style={linksStyle}>
+              {links.map((el) => (
+                <li style={{ marginRight: '20px' }} key={el.id}>
+                  <img src={el.icon} alt="icon" style={{ marginRight: '2.5px' }} />
+                  <a href={el.href} target="_blank" style={{ textDecoration: 'none' }}>
+                    {el.text}
+                  </a>
+                </li>
+              ))}
+            </ul>
```
Comment (AdammmVV) [inline src/components/MainFormPage/MainFormPage.tsx:57]: try use component [List](https://mantine.dev/core/list/#usage)

File: src/components/MainFormPage/MainFormPage.tsx [hunk 21]
```diff
@@ -0,0 +1,83 @@
...
+    dispatch(setDataForms(values));
+  };
+
+  return (
+    <Container style={container}>
+      <div style={headerContainer}>
+        <Box display="flex">
+          <Box style={avatar}>
+            <Text size="40px">РР</Text>
+          </Box>
```
Comment (AdammmVV) [inline src/components/MainFormPage/MainFormPage.tsx:43]: try use component [Avatar](https://mantine.dev/core/avatar/#usage)

File: src/components/MainFormPage/MainFormPage.tsx [hunk 22]
```diff
@@ -0,0 +1,83 @@
...
+          <Box>
+            <Text fw="500" size="20px">
+              Роман Романов
+            </Text>
+            <ul style={linksStyle}>
+              {links.map((el) => (
```
Comment (AdammmVV) [inline src/components/MainFormPage/MainFormPage.tsx:49]: use more informative name for `el ` or use destructuring

File: src/components/MainFormPage/MainFormPage.tsx [hunk 23]
```diff
@@ -0,0 +1,83 @@
...
+                <li style={{ marginRight: '20px' }} key={el.id}>
+                  <img src={el.icon} alt="icon" style={{ marginRight: '2.5px' }} />
```
Comment (AdammmVV) [inline src/components/MainFormPage/MainFormPage.tsx:51]: try user [Image](https://mantine.dev/core/image/#usage)

**Pull request merged**